cmd/hz/meta/manifest.go

```diff
@@ -39,36 +42,35 @@ func init() {
 	GoVersion, _ = gv.NewVersion(Version)
 }
 
-var regVersion = regexp.MustCompile(`hz version: v([0-9a-z.]*)`)
-
-func (manifest *Manifest) Validate(dir string) error {
-	rd, err := os.ReadFile(filepath.Join(dir, ManifestFile))
+func (manifest *Manifest) InitAndValidate(dir string) error {
+	m, err := loadConfigFile(filepath.Join(dir, ManifestFile))
 	if err != nil {
-		return err
+		return fmt.Errorf("can not load \".hz\", err: %v", err)
 	}
 
-	ms := regVersion.FindSubmatchIndex(rd)
-	if len(ms) != 4 || ms[2] < 19 {
-		logs.Errorf("Invalid manifest file: %s, pattern '%s' not match\n", filepath.Join(dir, ManifestFile), regVersion.String())
-		return fmt.Errorf("current project doesn't belong to hertz framework")
+	if len(m.Version) == 0 {
+		return fmt.Errorf("can not get hz version form \".hz\", current project doesn't belong to hertz framework")
 	}
-	v, err := gv.NewVersion(string(rd[ms[2]:ms[3]]))
+
+	*manifest = *m
+	_, err = gv.NewVersion(manifest.Version)
 	if err != nil {
-		return err
-	}
-	*manifest = Manifest{
-		Version: v,
+		return fmt.Errorf("invalid hz version in \".hz\", err: %v", err)
 	}
+
 	return nil
 }
 
-func (manifest Manifest) String() string {
-	return `// Code generated by hz. DO NOT EDIT.
+const hzTitle = "# Code generated by hz. DO NOT EDIT.\n"
```

Comment:
why modify "//" to "#"?  It will be a break change. The 'hzTitle' will be trimed when yaml unmarshal

Reply:
The annotator should not be "#" when in yaml format?

Reply:
That is why I  trimed 'hzTitle' when yaml unmarshal from '.hz'~ If you modify it, all project generated by previous 'hz' will failed to update😭😭😭 You can try it~

Reply:
> That is why I trimed 'hzTitle' when yaml unmarshal from '.hz'~ If you modify it, all project generated by previous 'hz' will failed to update😭😭😭 You can try it~

we could delete trim logic, and it can still work